src/components/movie-view/movie-view.jsx

```diff
@@ -1,28 +1,158 @@
+import { Link } from 'react-router-dom';
+
+// Import necessary Bootstrap components
+import Card from 'react-bootstrap/Card';
+import Button from 'react-bootstrap/Button';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+
+// Import the scss file for this view
+import './movie-view.scss';
 
 // Export the MovieView class that extends the component template React.Component
 export class MovieView extends React.Component {
+    constructor(props) {
+        super(props);
+        // Create state variables that will be used to add/remove a movie from a users favourites list
+        this.state = {
+            // REMOVED line of code, as I couldn't get an isFavourite flag to work as a state variable. My solution can be found in the render() function
+            // isFavourite: false,
+            favouriteMovies: [],
+            userDetails: []
+        }
+
+        // Bind these additional functions that will get called by onClick events to 'this'
+        this.addFavourite = this.addFavourite.bind(this);
+        this.removeFavourite = this.removeFavourite.bind(this);
+    }
+
+    // During componentDidMount() get the user's details (for displaying whether this movie is a favourite or not)
+    componentDidMount() {
+        let accessToken = localStorage.getItem('token');
+        this.getUserDetails(accessToken);
+    }
+
+    // getUserDetails function for making a request to the server for the users details
+    getUserDetails(token) {
+        axios.get(`https://t-dogg-movies-api.herokuapp.com/users/${this.props.user}`, {
+            headers: { Authorization: `Bearer ${token}`}
+        }).then(response => {
+            // Use the response to set the user details in the state variables
+            this.setState({
+                userDetails: response.data,
+                favouriteMovies: response.data.FavouriteMovies
+            });
+        }).catch(function(error) {
+            console.log(error);
+        });
+    }
+
+    // Function for adding this movie to a users favourites list. Makes a post request to the server using information passed in through the props
+    addFavourite() {
+        let token = localStorage.getItem('token');
+        // I'm not sure why I need the first {} (before the headers). but without those empty brackets all my requests returned unauthorized
+        axios.post(`https://t-dogg-movies-api.herokuapp.com/users/${this.props.user}/Movies/${this.props.movie._id}`, {}, {
+            headers: { Authorization: `Bearer ${token}` }
+        }).then(response => {
+            // Set the isFavourite state to true, now that this movie has been added to the list of favourites
+            // this.setState({ isFavourite: true });
+            // window.open refreshes the page to make sure this movie is correctly displaying as a favourite
+            window.open(`/movies/${this.props.movie._id}`, '_self');
```

Comment:
Don't refresh the page in an SPA. It is a really bad practise An SPA it's purpose is to have just one page that handles all routing and update data without reloading the page (and all the assets)
React should update the state accordingly instead. You need to mutate the state here